[lustre] Add Lustre filesystem plugin

utopiabound opened the pull request:
This collects some basic information about the configuration
of Lustre on either a client or a server, including:
* Interal ring buffer log
* Basic configuration information
* Quota information
* Configuration of LNet
* JobID info
* LDLM connection states

Signed-off-by: Nathaniel Clark <Nathaniel.Clark@misrule.us>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?

utopiabound commented:
This is the fixed version of Pull request #1057

bmr-cymru commented:
I'm just heading out for the day but based on a quick read this looks good - thanks for making the changes.

I see Adam's added his ack so I'll try to get a proper review done in the morning so we can go ahead and merge.

bmr-cymru commented on sos/plugins/lustre.py:
```diff
@@ -0,0 +1,50 @@
+    def setup(self):
+        self.add_cmd_output(["lctl debug_kernel",
+                             "lctl device_list"])
+
```
The conventional style in `sos` is as in the example I gave:

```python
    self.add_cmd_output([
        "command one",
        "command two",
        "command ..."
    ])
```

If you wouldn't mind respinning this and updating the branch, it helps to have as consistent a style throughout as we can (otherwise it's a minor tweak I can fix up when I merge).

utopiabound replied:
Sorry, I missed the formatting; I'll do that.

bmr-cymru commented on sos/plugins/lustre.py:
```diff
@@ -0,0 +1,50 @@
+
+        if (self.is_installed('lustre-client')):
+            self.add_cmd_output(["lfs df",
+                                 "lfs df -i"])
```
Ditto here.

bmr-cymru commented on sos/plugins/lustre.py:
```diff
@@ -0,0 +1,50 @@
+        else:
+            self.get_params("osd", ["osd-*.*.{mntdev,files*," +
+                                    "kbytes*,blocksize,brw_stats}"])
+            self.get_params("quota", ["osd-*.*.quota_slave." +
```
These are OK as they are; if anything we'd pregenerate the list in a variable, but it does not seem worth it here since there is no real commonality beyond the "osd-*.*." prefix.

bmr-cymru replied:
These may want to be protected by an `if not self.is_installed():` if the previous conditional is removed and these commands will spit error messages - in this case, because the command does exist on the system we will be able to run it, and will collect whatever output it generates in the report.

utopiabound commented:
This new patch should address all the previous comments.

utopiabound commented:
Is there anything else that's needed for this to land?

bmr-cymru commented on sos/plugins/lustre.py:
```diff
@@ -0,0 +1,54 @@
+        self.get_params("ldlm-states", ["*.*.state"])
+        self.get_params("jobid", ["jobid_name", "jobid_var"])
+
+        if (self.is_installed('lustre-client')):
```
There's no real reason to make this conditional - all command execution in `sos` is speculative (it's not a big deal, but it tends to make plugins cleaner/simpler if they follow the "forgiveness rather than permission" approach).

jgrund commented:
Is this able to land? Would like to have this in a sos report.

bmr-cymru commented:
We will try to pull it in for 3.5 (a few weeks overdue currently).

utopiabound commented:
I've updated the plugin with more information collection.  I removed the server/client distinction and just collect everything as that will be cleaner if for instance, the modules aren't installed via the "normal" rpms.

utopiabound commented:
This seems to have missed 3.5, can we land it for 3.6?

bmr-cymru commented:
> This seems to have missed 3.5, can we land it for 3.6?

It just got to the front of the review queue, so I hope so - we're expecting to release 3.6 next month.
